src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,115 @@
+    static final Pattern EVENT_PATTERN = Pattern.compile(EVENT_KEY);
+
+    public Parser() { }
+
+    private int findIndex(Pattern pattern, String input) {
+        Matcher matcher = pattern.matcher(input);
+        matcher.find();
+        return Integer.parseInt(matcher.group(3)) - 1;
+    }
+
+    private String findDescription(Pattern pattern, String input) {
+        Matcher matcher = pattern.matcher(input);
+        matcher.find();
+        return matcher.group(2).trim();
+    }
+
+    private String findEndTime(Pattern pattern, String input) {
+        Matcher matcher = pattern.matcher(input);
+        matcher.find();
+        return matcher.group(4).trim();
+    }
+
+    private static boolean isExitKey(String input) { return EXIT_KEY.equals(input); }
+
+    private static boolean isViewListKey(String input) {return VIEW_LIST_KEY.equals(input); }
+
+    private static boolean isDeleteKey(String input) {
+        Matcher deleteMatcher = DELETE_PATTERN.matcher(input);
+        return deleteMatcher.find();
+    }
+
+    private static boolean isFinishKey(String input) {
+        Matcher finishMatcher = FINISH_PATTERN.matcher(input);
+        return finishMatcher.find();
+    }
+
+    private static boolean isTodoKey(String input) {
+        Matcher todoMatcher = TODO_PATTERN.matcher(input);
+        return todoMatcher.find();
+    }
+
+    private static boolean isDeadLineKey(String input) {
+        Matcher deadlineMatcher = DEADLINE_PATTERN.matcher(input);
+        return deadlineMatcher.find();
+    }
+
+    private static boolean isEventKey(String input) {
+        Matcher eventMatcher = EVENT_PATTERN.matcher(input);
+        return eventMatcher.find();
+    }
+
+    public Command parseCommand(String userInput) throws NoCommandException {
+        if (this.isExitKey(userInput)) {
+            return new ExitCommand();
+        }
+        else if (this.isViewListKey(userInput)) {
+            return new ViewListCommand();
+        }
+        else if (this.isDeleteKey(userInput)) {
+            int deletedTaskIndex = this.findIndex(DELETE_PATTERN, userInput);
+            return new DeleteCommand(deletedTaskIndex);
+        }
+        else if (this.isFinishKey(userInput)) {
+            int finishedTaskIndex = this.findIndex(FINISH_PATTERN, userInput);
+            return new FinishCommand(finishedTaskIndex);
+        }
+        else if (this.isTodoKey(userInput)){
+            String description = this.findDescription(TODO_PATTERN, userInput);
+            ToDoTask taskToAdd = new ToDoTask();
+            return new AddCommand(taskToAdd, description);
+        }
+        else if (this.isDeadLineKey(userInput)){
+            String description = this.findDescription(DEADLINE_PATTERN, userInput);
+            String by = this.findEndTime(DEADLINE_PATTERN, userInput);
+            DeadLineTask taskToAdd = new DeadLineTask();
+            return new AddCommand(taskToAdd, description, by);
+        }
```

Comment:
This part can be handled using the chain or responsibility pattern. Allows you to minimize the length of the if / else blocks in the code.

FYI: https://refactoring.guru/design-patterns/chain-of-responsibility